### ISSUE TITLE
TypeError: no implicit conversion of String into Integer

### ISSUE THREAD

**Issue by chuichui9527** [Shopify/shopify-cli#1971]
# Issue summary

Write a short description of the issue here ↓

## Expected behavior

What do you think should happen?

## Actual behavior

What actually happens?

Tip: include an error message (in a `<details></details>` tag) if your issue is related to an error

## Steps to reproduce the problem

1.
1.
1.

## Stacktrace

C:/Ruby30-x64/lib/ruby/gems/3.0.0/gems/shopify-cli-2.10.0/lib/shopify_cli/theme/file.rb:113:in `[]='
C:/Ruby30-x64/lib/ruby/gems/3.0.0/gems/shopify-cli-2.10.0/lib/shopify_cli/theme/file.rb:113:in `visit_value'
C:/Ruby30-x64/lib/ruby/gems/3.0.0/gems/shopify-cli-2.10.0/lib/shopify_cli/theme/file.rb:106:in `block in visit_hash'
C:/Ruby30-x64/lib/ruby/gems/3.0.0/gems/shopify-cli-2.10.0/lib/shopify_cli/theme/file.rb:105:in `each'
C:/Ruby30-x64/lib/ruby/gems/3.0.0/gems/shopify-cli-2.10.0/lib/shopify_cli/theme/file.rb:105:in `visit_hash'

## Specifications

- App type:
- Operating System:
- Shell:
- Ruby version (ruby -v):

**Comment by amcaplan** [comment]
It looks like `visit_hash` is getting a hash with an Array value.

@macournoyer I assume you'd know the most about what's supposed to happen in this code and why it exists... sounds like a template file has some invalid configuration structure.

@chuichui9527 any additional information you could provide here would probably help a great deal in expediting finding a solution.

**Comment by amcaplan** [comment]
Looks like #1914 has the same problem.

**Comment by rmcfadzean** [comment]
Chiming in because I finally managed to fix this after hitting it today.

## tl;dr my fix

I had a template with 

```json
{
  "sections": {
    "order": []
  }
}
```

instead of

```json
{
  "sections": {},
  "order": []
}
```

## extra info

### the offending code (I think?)

https://github.com/Shopify/shopify-cli/blob/main/lib/shopify_cli/admin_api.rb#L84 was actually throwing an error but somehow being swallowed up by the error handling part of the kit (this was a big headache to combat. I ended up spending 99% of my time fighting it with `puts` debugging...)

I also had to overwrite the hash comparison in https://github.com/Shopify/shopify-cli/blob/main/lib/shopify_cli/theme/syncer.rb#L327 to force the file to attempt an update.

In hindsight this may actually be a bit of a red herring and the error could be in the checksum creation or before it and I was just bruteforcing by.

### stack trace

```
/home/rob/.asdf/installs/ruby/3.1.0/lib/ruby/gems/3.1.0/gems/shopify-cli-2.13.0/lib/shopify_cli/theme/file.rb:117:in `[]=': no implicit conversion of String into Integer (TypeError)
        from /home/rob/.asdf/installs/ruby/3.1.0/lib/ruby/gems/3.1.0/gems/shopify-cli-2.13.0/lib/shopify_cli/theme/file.rb:117:in `visit_value'
        from /home/rob/.asdf/installs/ruby/3.1.0/lib/ruby/gems/3.1.0/gems/shopify-cli-2.13.0/lib/shopify_cli/theme/file.rb:110:in `block in visit_hash'
        from /home/rob/.asdf/installs/ruby/3.1.0/lib/ruby/gems/3.1.0/gems/shopify-cli-2.13.0/lib/shopify_cli/theme/file.rb:109:in `each'
        from /home/rob/.asdf/installs/ruby/3.1.0/lib/ruby/gems/3.1.0/gems/shopify-cli-2.13.0/lib/shopify_cli/theme/file.rb:109:in `visit_hash'
        from /home/rob/.asdf/installs/ruby/3.1.0/lib/ruby/gems/3.1.0/gems/shopify-cli-2.13.0/lib/shopify_cli/theme/file.rb:104:in `visit_document'
        from /home/rob/.asdf/installs/ruby/3.1.0/lib/ruby/gems/3.1.0/gems/shopify-cli-2.13.0/lib/shopify_cli/theme/file.rb:93:in `normalize_json'
        from /home/rob/.asdf/installs/ruby/3.1.0/lib/ruby/gems/3.1.0/gems/shopify-cli-2.13.0/lib/shopify_cli/theme/file.rb:73:in `checksum'
        from /home/rob/.asdf/installs/ruby/3.1.0/lib/ruby/gems/3.1.0/gems/shopify-cli-2.13.0/lib/shopify_cli/theme/syncer.rb:328:in `file_has_changed?'
        from /home/rob/.asdf/installs/ruby/3.1.0/lib/ruby/gems/3.1.0/gems/shopify-cli-2.13.0/lib/shopify_cli/theme/syncer.rb:203:in `enqueue'
        from /home/rob/.asdf/installs/ruby/3.1.0/lib/ruby/gems/3.1.0/gems/shopify-cli-2.13.0/lib/shopify_cli/theme/syncer.rb:58:in `block in enqueue_updates'
        from /home/rob/.asdf/installs/ruby/3.1.0/lib/ruby/gems/3.1.0/gems/shopify-cli-2.13.0/lib/shopify_cli/theme/syncer.rb:58:in `each'
        from /home/rob/.asdf/installs/ruby/3.1.0/lib/ruby/gems/3.1.0/gems/shopify-cli-2.13.0/lib/shopify_cli/theme/syncer.rb:58:in `enqueue_updates'
        from /home/rob/.asdf/installs/ruby/3.1.0/lib/ruby/gems/3.1.0/gems/shopify-cli-2.13.0/lib/shopify_cli/theme/syncer.rb:146:in `upload_theme!'
        from /home/rob/.asdf/installs/ruby/3.1.0/lib/ruby/gems/3.1.0/gems/shopify-cli-2.13.0/lib/project_types/theme/ui/sync_progress_bar.rb:11:in `public_send'
        from /home/rob/.asdf/installs/ruby/3.1.0/lib/ruby/gems/3.1.0/gems/shopify-cli-2.13.0/lib/project_types/theme/ui/sync_progress_bar.rb:11:in `block in progress'
        from /home/rob/.asdf/installs/ruby/3.1.0/lib/ruby/gems/3.1.0/gems/shopify-cli-2.13.0/vendor/deps/cli-ui/lib/cli/ui/progress.rb:34:in `progress'
        from /home/rob/.asdf/installs/ruby/3.1.0/lib/ruby/gems/3.1.0/gems/shopify-cli-2.13.0/lib/project_types/theme/ui/sync_progress_bar.rb:10:in `progress'
        from /home/rob/.asdf/installs/ruby/3.1.0/lib/ruby/gems/3.1.0/gems/shopify-cli-2.13.0/lib/project_types/theme/commands/serve.rb:26:in `block in call'
        from /home/rob/.asdf/installs/ruby/3.1.0/lib/ruby/gems/3.1.0/gems/shopify-cli-2.13.0/lib/shopify_cli/theme/dev_server.rb:53:in `block in start'
        from /home/rob/.asdf/installs/ruby/3.1.0/lib/ruby/gems/3.1.0/gems/shopify-cli-2.13.0/vendor/deps/cli-ui/lib/cli/ui/frame.rb:103:in `open'
        from /home/rob/.asdf/installs/ruby/3.1.0/lib/ruby/gems/3.1.0/gems/shopify-cli-2.13.0/lib/shopify_cli/theme/dev_server.rb:49:in `start'
        from /home/rob/.asdf/installs/ruby/3.1.0/lib/ruby/gems/3.1.0/gems/shopify-cli-2.13.0/lib/project_types/theme/commands/serve.rb:25:in `call'
        from /home/rob/.asdf/installs/ruby/3.1.0/lib/ruby/gems/3.1.0/gems/shopify-cli-2.13.0/lib/shopify_cli/command/sub_command.rb:16:in `call'
        from /home/rob/.asdf/installs/ruby/3.1.0/lib/ruby/gems/3.1.0/gems/shopify-cli-2.13.0/lib/shopify_cli/command.rb:27:in `call'
        from /home/rob/.asdf/installs/ruby/3.1.0/lib/ruby/gems/3.1.0/gems/shopify-cli-2.13.0/lib/shopify_cli/core/executor.rb:17:in `block (2 levels) in call'
        from /home/rob/.asdf/installs/ruby/3.1.0/lib/ruby/gems/3.1.0/gems/shopify-cli-2.13.0/vendor/deps/cli-kit/lib/cli/kit/executor.rb:37:in `block (2 levels) in with_logging'
        from /home/rob/.asdf/installs/ruby/3.1.0/lib/ruby/gems/3.1.0/gems/shopify-cli-2.13.0/vendor/deps/cli-ui/lib/cli/ui/stdout_router.rb:169:in `with_id'
        from /home/rob/.asdf/installs/ruby/3.1.0/lib/ruby/gems/3.1.0/gems/shopify-cli-2.13.0/vendor/deps/cli-kit/lib/cli/kit/executor.rb:36:in `block in with_logging'
        from /home/rob/.asdf/installs/ruby/3.1.0/lib/ruby/gems/3.1.0/gems/shopify-cli-2.13.0/vendor/deps/cli-ui/lib/cli/ui.rb:176:in `log_output_to'
        from /home/rob/.asdf/installs/ruby/3.1.0/lib/ruby/gems/3.1.0/gems/shopify-cli-2.13.0/vendor/deps/cli-kit/lib/cli/kit/executor.rb:35:in `with_logging'
        from /home/rob/.asdf/installs/ruby/3.1.0/lib/ruby/gems/3.1.0/gems/shopify-cli-2.13.0/lib/shopify_cli/core/executor.rb:16:in `block in call'
        from /home/rob/.asdf/installs/ruby/3.1.0/lib/ruby/gems/3.1.0/gems/shopify-cli-2.13.0/vendor/deps/cli-kit/lib/cli/kit/executor.rb:45:in `block (2 levels) in with_traps'
        from /home/rob/.asdf/installs/ruby/3.1.0/lib/ruby/gems/3.1.0/gems/shopify-cli-2.13.0/vendor/deps/cli-kit/lib/cli/kit/executor.rb:51:in `twrap'
        from /home/rob/.asdf/installs/ruby/3.1.0/lib/ruby/gems/3.1.0/gems/shopify-cli-2.13.0/vendor/deps/cli-kit/lib/cli/kit/executor.rb:44:in `block in with_traps'
        from /home/rob/.asdf/installs/ruby/3.1.0/lib/ruby/gems/3.1.0/gems/shopify-cli-2.13.0/vendor/deps/cli-kit/lib/cli/kit/executor.rb:55:in `twrap'
        from /home/rob/.asdf/installs/ruby/3.1.0/lib/ruby/gems/3.1.0/gems/shopify-cli-2.13.0/vendor/deps/cli-kit/lib/cli/kit/executor.rb:43:in `with_traps'
        from /home/rob/.asdf/installs/ruby/3.1.0/lib/ruby/gems/3.1.0/gems/shopify-cli-2.13.0/lib/shopify_cli/core/executor.rb:15:in `call'
        from /home/rob/.asdf/installs/ruby/3.1.0/lib/ruby/gems/3.1.0/gems/shopify-cli-2.13.0/lib/shopify_cli/core/entry_point.rb:24:in `block in call'
        from /home/rob/.asdf/installs/ruby/3.1.0/lib/ruby/gems/3.1.0/gems/shopify-cli-2.13.0/lib/shopify_cli/core/monorail.rb:30:in `log'
        from /home/rob/.asdf/installs/ruby/3.1.0/lib/ruby/gems/3.1.0/gems/shopify-cli-2.13.0/lib/shopify_cli/core/entry_point.rb:23:in `call'
        from /home/rob/.asdf/installs/ruby/3.1.0/lib/ruby/gems/3.1.0/gems/shopify-cli-2.13.0/bin/shopify:50:in `block (2 levels) in <top (required)>'
        from /home/rob/.asdf/installs/ruby/3.1.0/lib/ruby/gems/3.1.0/gems/shopify-cli-2.13.0/vendor/deps/cli-kit/lib/cli/kit/error_handler.rb:75:in `handle_abort'
        from /home/rob/.asdf/installs/ruby/3.1.0/lib/ruby/gems/3.1.0/gems/shopify-cli-2.13.0/vendor/deps/cli-kit/lib/cli/kit/error_handler.rb:21:in `call'
        from /home/rob/.asdf/installs/ruby/3.1.0/lib/ruby/gems/3.1.0/gems/shopify-cli-2.13.0/bin/shopify:49:in `block in <top (required)>'
        from /home/rob/.asdf/installs/ruby/3.1.0/lib/ruby/gems/3.1.0/gems/shopify-cli-2.13.0/bin/shopify:59:in `<top (required)>'
        from /home/rob/.asdf/installs/ruby/3.1.0/bin/shopify:25:in `load'
        from /home/rob/.asdf/installs/ruby/3.1.0/bin/shopify:25:in `<main>'
```
